x-pack/plugins/fleet/server/services/epm/elasticsearch/template/template.ts

```diff
@@ -642,7 +645,25 @@ const updateExistingDataStream = async ({
   esClient: ElasticsearchClient;
   logger: Logger;
 }) => {
+  const dataStreamResponse = await esClient.indices.getDataStream({ name: dataStreamName });
+  const currentBackingIndex =
+    dataStreamResponse.data_streams[0].indices[
+      dataStreamResponse.data_streams[0].indices.length - 1
+    ];
+
+  const currentBackingIndexData = (
+    await esClient.indices.get({
+      index: currentBackingIndex.index_name,
+    })
+  )[currentBackingIndex.index_name];
+
+  const currentIndexMode = currentBackingIndexData.settings?.index?.mode;
+  // @ts-expect-error
+  const currentSourceType = currentBackingIndexData.mappings?._source?.mode;
```

Comment:
Is this ts error because of mode doesn't exist on MappingSourceField? Interesting why the es type doesn't have this field.

Reply:
Yeah the ES types don't have `_source.mode` defined - not sure where to contribute that change but it's been the case for a few releases now.